nemo_rl/models/policy/megatron_policy_worker.py

```diff
@@ -920,32 +950,79 @@ def get_logprobs(
         pp_grp = get_pipeline_model_parallel_group()
         pp_size = get_pipeline_model_parallel_world_size()
 
+        # if pp_size > 1, we need to pad the full sequence to the max sequence length to maintain a static PP buffer
+        if (
+            self.cfg["sequence_packing"]["enabled"]
+            and self.cfg["megatron_cfg"]["pipeline_model_parallel_size"] > 1
+        ):
+            _, pad_full_seq_to = (
+                data.get_microbatch_iterator_for_packable_sequences_len()
+            )
+            pp_seq_dim_size = pad_full_seq_to
+        else:
+            pad_full_seq_to = None
+
         def forward_step_fn(
             data_iterator: Iterator[BatchedDataDict[Any]], model: GPTModel
         ):
+            nonlocal pad_full_seq_to
             data_dict = next(data_iterator).to("cuda")
-            input_ids = data_dict["input_ids"]
-            attention_mask, _, position_ids = get_ltor_masks_and_position_ids(
-                input_ids, 0, False, False, False
-            )
+            if self.cfg["sequence_packing"]["enabled"]:
+                original_seq_length = data_dict["input_ids"].shape[1]
+                tp_size = self.cfg["megatron_cfg"]["tensor_model_parallel_size"]
+                pp_size = self.cfg["megatron_cfg"]["pipeline_model_parallel_size"]
+                cp_size = self.cfg["megatron_cfg"]["context_parallel_size"]
+                pad_factor = cp_size * 2 * tp_size if cp_size > 1 else tp_size
+                input_ids, packed_seq_params, cu_seqlens, cu_seqlens_padded = (
+                    _pack_sequences_for_megatron(
+                        data_dict["input_ids"].clone(),
+                        data_dict["input_lengths"],
+                        pad_individual_seqs_to_multiple_of=pad_factor,
+                        pad_packed_seq_to=pad_full_seq_to,
+                    )
+                )
+                input_ids = input_ids
```

Comment:
@SahilJain314 Do you mind taking another look at this code block.

`input_ids = input_ids` doesn't make sense